### PR TITLE
Static site generation optimisation

### PR DESCRIPTION
**Pull request by domderen**
While preparing to exclude your this static page generation i thought of this small optimisation, and I thought you guys might be interested in it.

I am sending this PR mainly for the purposes of the peer review to see if I didn't miss any reasons why you used express in your static page generation process,
and to see if it could indeed be skipped, like I did here.

Thanks in advance for any feedback.

Static page rendering optimisation
- Bypassing express server in static site generation,
- dump-html now uses the same render function, as the express middleware,
- removed fetchPage method, as it was now not required,
- simplified build-prod script to only run production web pack build, and html-dump, no express sever in the middle.

### CODE REVIEW

**Comment by harrison** [comment]
Nice idea! Haven't had a chance to properly look this over yet, will do it in the next couple of days.

File: server/render.js [hunk 1]
```diff
@@ -27,69 +27,90 @@ function normalizeUrl(urlStr) {
...
+        <HtmlDocument
+          markup={markup}
+          script={webpackUrls.script}
+          css={webpackUrls.css}
+          router={router}
+          dataRender={appProps(stateProps)}
+          {...appProps(stateProps)} />
+      );
+      const doctype = '<!DOCTYPE html>';
+      resolve([requestUrl, doctype + html]);
```
Comment (harrison) [inline server/render.js]: think we can just do `resolve(doctype + html)` - this format is used by `writeHtml` but would be better to deal with it in `dump-html`

File: server/render.js [hunk 2]
```diff
@@ -27,69 +27,90 @@ function normalizeUrl(urlStr) {
...
-        script={webpackUrls.script}
-        css={webpackUrls.css}
-        router={router}
-        dataRender={appProps(stateProps)}
-        {...appProps(stateProps)} />
-    );
-    const doctype = '<!DOCTYPE html>';
-    res.send(doctype + html);
+  render(req.url)
+  .then(([, htmlOutput]) => {
```
Comment (harrison) [inline server/render.js]: `.then((htmlOutput) => {`

File: script/dump-html [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
-  console.error(`Usage: ${bin} ${script} host directory`);
+if (!outDir) {
+  console.error(`Usage: ${bin} ${script} directory`);
   process.exit(1);
 }
 
 const processPage = function(pagePath) {
-  return fetchPage(serverUrl, pagePath)
+  return render(pagePath)
     .then(writePage.bind(null, outDir))
```
Comment (harrison) [inline script/dump-html]: `.then((htmlContent) => { return writePage(outDir, [pagePath, htmlContent]); })`

File: app/static-build/builder.spec.js [hunk 4]
```diff
@@ -1,40 +1,14 @@
-import nock from 'nock';
```
Comment (harrison) [inline app/static-build/builder.spec.js]: could remove `nock` from the package.json

File: script/build-prod [hunk 5]
```diff
@@ -1,35 +1,18 @@
...
-
-./script/dump-html "http://localhost:$PORT" 'dist'
+./script/dump-html 'dist'
 DUMP_EXIT=$?
 
-kill $SERVER_PID
-
 set -e errexit
 
 if [ $DUMP_EXIT -eq 0 ]; then
```
Comment (harrison) [inline script/build-prod]: think this needs to be resolved in `script/build` now

**Comment by megaslug** [comment]
Closing to address @harrison comments [here](https://github.com/gocardless/splash-pages/pull/837)

**Pull request closed without merging**